provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -812,7 +812,7 @@ func (c *Client) cloneVM(
 	}
 	info, err := taskWaiter.waitTask(ctx, task, "cloning VM")
 	if err != nil {
-		return nil, errors.Annotatef(err, "waiting for task %q", info.Name)
+		return nil, errors.Annotatef(err, "waiting for task")
```

Comment:
It'd still be nice to have some detail about which task we were waiting on, machines are provisioned in parallel. 

```suggestion
		return nil, errors.Annotatef(err, "waiting on cloneVM task for %q", args.Name)
```